sonar-cxx-plugin/src/test/java/org/sonar/plugins/cxx/TestUtils.java

```diff
@@ -171,16 +172,16 @@ public static void addInputFile(DefaultFileSystem fs,
         temp.setCharAt(0, Character.toUpperCase(temp.charAt(0)));
         path = temp.toString();
       }
-      inputFile = new DefaultInputFile(path);
-      inputFile.setAbsolutePath(path);
+      inputFile = new DefaultInputFile("key", path);
```

Comment:
and what about if we just mock everything related with filesystem, in the end we want to test the parser not really the file system. if doing it like this then test passes fine